src/Symfony/Component/DependencyInjection/Compiler/ServiceLocatorTagPass.php

```diff
@@ -72,10 +72,13 @@ protected function processValue($value, $isRoot = false)
     /**
      * @param ContainerBuilder $container
      * @param Reference[]      $refMap
+     * @param string|null      $callerId
      *
      * @return Reference
+     *
+     * @final since version 3.4
      */
-    public static function register(ContainerBuilder $container, array $refMap)
+    public static function register(ContainerBuilder $container, array $refMap/*, string $callerId = null*/)
```

Comment:
the class is final (not even `@final` but actually final), so you can add the new argument without this magic